hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoBatchSqlConnectorTest.java

```diff
@@ -135,6 +137,89 @@ public void readWithTypeCoertion() {
         );
     }
 
+    @Test
+    public void readWithOneProcessor() {
+        MongoCollection<Document> collection = database.getCollection(collectionName);
+        collection.insertOne(new Document("firstName", "Luke").append("lastName", "Skywalker").append("jedi", "true"));
+
+        Pattern parallelismDagPattern = Pattern.compile("\\[localParallelism=(\\d)+]");
+        AtomicBoolean otherFound = new AtomicBoolean(false);
+        LogListener lookForProjectAndFilterStep = log -> {
+            String message = log.getLogRecord().getMessage();
+            Matcher matcher = parallelismDagPattern.matcher(message);
+            if (matcher.find()) {
+                String number = matcher.group(1);
+                if (!"1".equals(number)) {
+                    otherFound.set(true);
+                }
+            }
+        };
+        for (HazelcastInstance instance : instances()) {
```

Comment:
have you tested it with >1 instance? `MongoSqlTest` creates 1 HZ member

Reply:
[2 ](https://github.com/hazelcast/hazelcast/blob/2e48297cf26d9938d0d805f8f6beb12803318a33/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/mongodb/MongoSqlTest.java#L58) :)

Reply:
too many concurrent PRs ;)